openstack/module_defaults

```diff
@@ -54,7 +54,7 @@ MOD_PARAMS[__PATH_MTU__]=
 
 # This is enough for creating one ubuntu vm or multiple cirros vms but some
 # scenarios may want to allow more per compute (e.g. octavia).
-MOD_PARAMS[__NOVA_COMPUTE_UNIT_CONSTRAINTS__]="mem=4G root-disk=80G"
+MOD_PARAMS[__NOVA_COMPUTE_UNIT_CONSTRAINTS__]="mem=4G"
```

Comment:
Im afraid this will break octavia which needs enough capacity to spawn an amphora vm and a member vm.

Reply:
The VMs are stored on the ephemeral storage disk, which has in practice almost always been attached at a size of 50GB anytime you used Vault which is most of the time as it's pulled in by OVN. This is due to `stoorage: ephemeral-device: cinder,50G,1` in overlays/openstack/vault-openstack-secrets.yaml.

The storage space from this mount point is what is reflected in the placement service/hypervisor stats when it's mounted, hence the 80GB root-disk size has mostly been irrelevant.

To handle non-OVN environments (either explicitly requested or for very old openstack versions where we didn't default to it), I also moved the `stroage: ephemeral-device: cinder,50G,1` from overlays/openstack/vault-openstack-secrets.yaml to openstack.yaml.template so that we would also always have this extra disk space in any deployment.

Reply:
my deployment seems to not be behaving that way. I have vault related to nova-compute:secret-storage but I still have an 80G root volume:

```
$ juju ssh nova-compute/0 df -h| egrep '/dev/vda1 '
/dev/vda1        78G  3.3G   75G   5% /
```

Reply:
I assume that juju is picking the greater of the two constraints

Reply:
I see your point now, nova is using the space available in /varlib/nova/instances not /